src/main/scala/org/scalajs/cli/Scalajsp.scala

```diff
@@ -91,24 +98,49 @@ object Scalajsp {
     throw new AssertionError("unreachable")
   }
 
-  private def readFromFile(fileName: String): VirtualScalaJSIRFile = {
+  private def readFromFile(fileName: String): Future[VirtualScalaJSIRFile] = {
     val file = new File(fileName)
 
-    if (!file.exists)
+    if (!file.exists) {
       fail(s"No such file: $fileName")
-    else if (!file.canRead)
+    } else if (!file.canRead) {
       fail(s"Unable to read file: $fileName")
-    else
-      new FileVirtualScalaJSIRFile(file, file.getName)
+    } else {
+      for {
+        container <- FileScalaJSIRContainer.fromSingleFile(file.toPath())
+        sjsirFiles <- container.sjsirFiles
+      } yield {
+        sjsirFiles.head
+      }
```

Comment:
Ouch... This is clearly a shortcoming of the API as it stands now. [scala-js#3675](https://github.com/scala-js/scala-js/pull/3675/files#diff-eef0914ccc351a8cb0fd2c489baf4755) should fix this though.